### PR TITLE
Add comprehensive tests for Unlock feature

### PR DESCRIPTION
**Pull request by SaintPatrck**
## 🎟️ Tracking

Test coverage improvements for critical unlock/authentication functionality.

## 📔 Objective

This PR adds comprehensive test coverage for the Unlock feature, which serves as the authentication gate for the Authenticator application. The unlock flow is security-critical and requires thorough testing to ensure correct behavior across all user scenarios.

**Test Coverage:**
- **UnlockViewModel**: Complete test suite covering authentication state management, biometric flows, PIN validation, and error handling
- **UnlockScreen**: UI test coverage for Compose screen rendering, user interactions, and state-driven UI updates

**Critical Scenarios Tested:**
- Biometric authentication success/failure flows
- PIN entry validation and verification
- Account unlock state transitions
- Error handling and user feedback
- Edge cases: empty PIN, biometric unavailable, authentication timeouts

The unlock screen is the first security barrier users encounter. These tests ensure authentication mechanisms function correctly, handle failures gracefully, and maintain proper security posture across all scenarios.

## 📸 Screenshots

N/A - Test-only changes

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: authenticator/src/test/kotlin/com/bitwarden/authenticator/ui/auth/unlock/UnlockScreenTest.kt [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+        verify {
+            mockViewModel.trySendAction(UnlockAction.BiometricsUnlockClick)
+        }
+    }
+
+    @Test
+    fun `NavigateToItemListing event should call onUnlocked callback`() {
+        mutableEventFlow.tryEmit(UnlockEvent.NavigateToItemListing)
+
+        assert(onUnlockedCalled)
```
Comment (david-livefront) [inline authenticator/src/test/kotlin/com/bitwarden/authenticator/ui/auth/unlock/UnlockScreenTest.kt]: Can we use `assertTrue`

File: authenticator/src/test/kotlin/com/bitwarden/authenticator/ui/auth/unlock/UnlockViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+
+        viewModel.stateFlow.test {
+            skipItems(1) // Initial state
+
+            viewModel.trySendAction(UnlockAction.BiometricsUnlockSuccess(mockCipher))
+
+            skipItems(1) // Loading state
+            val finalState = awaitItem()
+
+            assertEquals(null, finalState.dialog)
```
Comment (david-livefront) [inline authenticator/src/test/kotlin/com/bitwarden/authenticator/ui/auth/unlock/UnlockViewModelTest.kt]: Can we assert the whole state

File: authenticator/src/test/kotlin/com/bitwarden/authenticator/ui/auth/unlock/UnlockViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+                title = BitwardenString.an_error_has_occurred.asText(),
+                message = BitwardenString.generic_error_message.asText(),
+                throwable = testError,
+            ),
+        )
+        assertEquals(expected, state)
+    }
+
+    private fun createViewModel(
+        savedStateHandle: SavedStateHandle = SavedStateHandle(),
```
Comment (david-livefront) [inline authenticator/src/test/kotlin/com/bitwarden/authenticator/ui/auth/unlock/UnlockViewModelTest.kt]: We try to avoid passing in a `SavedStateHandle` in the `createViewModel` methods in favor of passing in a nullable state instead. It makes things much clearer

File: authenticator/src/test/kotlin/com/bitwarden/authenticator/ui/auth/unlock/UnlockScreenTest.kt [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+
+    @Before
+    fun setUp() {
+        onUnlockedCalled = false
+        setContent(
+            biometricsManager = mockBiometricsManager,
+        ) {
+            UnlockScreen(
+                viewModel = mockViewModel,
+                biometricsManager = mockBiometricsManager,
```
Comment (david-livefront) [inline authenticator/src/test/kotlin/com/bitwarden/authenticator/ui/auth/unlock/UnlockScreenTest.kt]: Not a huge deal but you don't need this `biometricsManager` here since it's in the `setContent`

File: authenticator/src/test/kotlin/com/bitwarden/authenticator/ui/auth/unlock/UnlockScreenTest.kt [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+    }
+
+    @Test
+    fun `loading dialog should display when state has Loading dialog`() {
+        mutableStateFlow.value = DEFAULT_STATE.copy(
+            dialog = UnlockState.Dialog.Loading,
+        )
+
+        composeTestRule
+            .onNodeWithText("Loading")
```
Comment (david-livefront) [inline authenticator/src/test/kotlin/com/bitwarden/authenticator/ui/auth/unlock/UnlockScreenTest.kt:212]: For these dialog tests, should we assert that the text is being displayed in a dialog?

File: authenticator/src/test/kotlin/com/bitwarden/authenticator/ui/auth/unlock/UnlockViewModelTest.kt [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+                isBiometricsEnabled = true,
+                isBiometricsValid = true,
+                showBiometricInvalidatedMessage = false,
+                dialog = UnlockState.Dialog.Error(
+                    title = BitwardenString.an_error_has_occurred.asText(),
+                    message = BitwardenString.generic_error_message.asText(),
+                    throwable = testError,
+                ),
+            )
+            assertEquals(expected, errorState)
```
Comment (david-livefront) [inline authenticator/src/test/kotlin/com/bitwarden/authenticator/ui/auth/unlock/UnlockViewModelTest.kt]: Maybe I am being too picky but we can just do this inline.

```koltin
assertEquals(
    UnlockState(
        isBiometricsEnabled = true,
        isBiometricsValid = true,
        showBiometricInvalidatedMessage = false,
        dialog = UnlockState.Dialog.Error(
            title = BitwardenString.an_error_has_occurred.asText(),
            message = BitwardenString.generic_error_message.asText(),
            throwable = testError,
        ),
    ),
    awaitItem().
)
```

File: authenticator/src/test/kotlin/com/bitwarden/authenticator/ui/auth/unlock/UnlockViewModelTest.kt [hunk 7]
```diff
@@ -0,0 +1,385 @@
...
+        } returns BiometricsUnlockResult.InvalidStateError(testError)
+
+        val viewModel = createViewModel()
+
+        viewModel.stateFlow.test {
+            skipItems(1) // Initial state
+
+            viewModel.trySendAction(UnlockAction.BiometricsUnlockSuccess(mockCipher))
+
+            skipItems(1) // Loading state
```
Comment (david-livefront) [inline authenticator/src/test/kotlin/com/bitwarden/authenticator/ui/auth/unlock/UnlockViewModelTest.kt]: We could assert the loading state instead of skipping it?

```kotlin
assertEquals(
    startState.copy(dialog = UnlockState.Dialog.Error(Loading)),
    awaitItem(),
)
```

File: authenticator/src/test/kotlin/com/bitwarden/authenticator/ui/auth/unlock/UnlockScreenTest.kt [hunk 8]
```diff
@@ -0,0 +1,287 @@
...
+        every { eventFlow } returns mutableEventFlow
+        every { trySendAction(any()) } just runs
+    }
+
+    private val mockBiometricsManager: BiometricsManager = mockk(relaxed = true)
+    private val mockCipher: Cipher = mockk()
+
+    @Before
+    fun setUp() {
+        onUnlockedCalled = false
```
Comment (david-livefront) [inline authenticator/src/test/kotlin/com/bitwarden/authenticator/ui/auth/unlock/UnlockScreenTest.kt]: You should not need this
